tests/test_europy.py

```diff
@@ -104,9 +104,53 @@ def test_model_details(details: ModelDetails=None):
         assert loaded_details.title == details.title
         assert loaded_details.description != details.description
 
-    assert True
+@model_details('tests/model_details_example.yml')
+def test_model_details_ymal(details: ModelDetails=None):
+    import yaml
+    details.description += '... this is computed yaml description'
+
+    with open('tests/model_details_example.yml', 'r') as f:
+        loaded_details = ModelDetails(**yaml.load(f, Loader=yaml.FullLoader))
+
+        assert loaded_details.title == details.title
+        assert loaded_details.description != details.description
+
 
 # this must run in order to pass
 @model_details() # this will load the latest in the report
 def test_loaded_model_details(details: ModelDetails=None):
-    assert '... this is a computed description' in details.description
+    assert '... this is computed yaml description' in details.description
+
+
+@using_params('tests/param_example.yml')
+def test_params(op1: int=None, op2: int=None, text_example: str=None, list_example: List[float]=[], a_global_param=None):
+    assert op1 != None, "op1 should be populated from params"
+    assert op2 != None, "op1 should be populated from params"
+    assert text_example != None, "text_example should be populated from params"
+    assert list_example != [], "list_example should be populated from params"
+    assert a_global_param != None, "a_global_param should be populated from params"
+
+
+@report_plt("example_figure")
+def test_save_image():
```

Comment:
Interesting, I was thinking of just adapting the "collector" to do something different if the result of this function is a plot. Instead of adding a new decorator because now I think it would get funky if we added `@bias` since it would make it dependent on the ordering. We could also easily follow up on this later, my thought would be that `ReportFigure` or `plt` would be one of the types that `TestResult.result` could take on (in addition to its current union of types).

Reply:
Yeah this was my initial thought too, but didn't want to mess with anything you are working on. I figure there could be a use for this scenario and then we can integrate into the TestResult.